docs/blueprints/slow-down-and-retry.md

```diff
@@ -0,0 +1,87 @@
+```
+  begin object upload
+    (for remote segment)
+      begin segment upload 
+        send data to storage node
+      commit segment upload
+    (or for inline segment)
+      make inline segment
+    ... repeat for multiple segments
+  commit object upload
+```
+
+Requests are batched:
+1. begin object upload + (first begin segment upload or make inline segment)
+2. commit segment upload + (begin segment upload or make inline segment) for multi-segment object
+3. (last commit segment upload or last make inline segment) + commit object upload
+4. special case for multi-segment object with last segment inline: commit segment upload + last make inline segment + commit object upload
+
+In case of `Too Many Requests` during upload operation we should be able to retry batched requests as soon as we resolve point 1 from Open Issues. Other case for upload is when we fail to upload single segment to storage nodes then we should retry uploading whole segment from scratch. In future we can improve this logic to re-upload failed pieces to different storage nodes.
+
+In case of `Too Many Requests` during download operation we should be able to retry single Metainfo API requests. If downloading segment from storage nodes will fail we should retry downloading this single segment.
+
+As an addition to retry logic we need to implement at least basic backoff mechanism that will increase the waiting time (between retries) up to a certain threshold. This will prevent from overwhelming satellite with retry requests. For our needs exponential backoff looks to be reasonable solution. 
+
+As an alternative for implementing exponential backoff on client side we can also consider to return with `To Many Request` error information when client should try to repeat request. The HTTP protocol defines that is such situation server may include a `Retry-After` header ([429](https://httpstatuses.com/429)). We can use similar approach and return delay value more attached to current satellite load.
+
+Additional rules:
+* Every retried request should be processed by backoff logic.
+* Retry and slow mechanism should be transient from libuplink consumer perspective except several necessary configurable options (see Implementation) and logging (see next point).
+* Every retry should be logged (severity WARNING or lower) to give details why retry occurs and how long it will take to repeat request.
+* Applying retry and slow down logic shouldn't degrade libuplink overall performance and increase number of round trips to satellite.
+
+## Implementation
+
+TODO: will be added when general design will be accepted.
+
+Options:
+* initial interval
+* max elapsed time or number of retries ??
+* multipliter
+* randomization factor
+* max interval
+
+## Open issues
+
+1. Currently, endpoints handled by DRPC in case of error are not returning value, only error. With Metainfo `Batch` call we would need to have partial results which requests were executed successfully. 
```

Comment:
Batch call should return one return value per every batch call, or the first one to fail returns an error.

Reply:
batch returns responses from all batch calls and in case of error it returns responses from successfully executed calls + error